lib/src/peppermint_connect.dart

```diff
@@ -0,0 +1,365 @@
+import 'dart:convert';
+
+import 'package:eth_sig_util/eth_sig_util.dart';
+import 'package:flutter_secure_storage/flutter_secure_storage.dart';
+import 'package:wallet_connect/wallet_connect.dart';
+import 'package:web3dart/crypto.dart';
+import 'package:web3dart/web3dart.dart';
+import 'package:http/http.dart' as http;
+
+class WCAttributes {
+  WCClient wcClient;
+  WCSessionStore? sessionStore;
+  Web3Client? web3client;
+
+  WCAttributes(this.wcClient, this.sessionStore, this.web3client);
+}
+
+class WalletConnectManager {
+  /// Use Flutter Secure Storage to safely store keys.
+  ///
+  /// See documentation https://pub.dev/packages/flutter_secure_storage
+  final storage = const FlutterSecureStorage();
+
+  /// Initialize Wallet Connect
+  ///
+  /// This function is used to initialize a Wallet Connect connection by providing
+  /// various callbacks to handle different connection-related events such as session requests,
+  /// failures, disconnections, Ethereum transaction signing, and Ethereum transaction sending.
+  /// The function also initializes the WCClient and Web3Client clients used to communicate
+  /// with the blockchain service provider.
+  ///
+  /// Parameters:
+  /// - onSessionRequest: A callback invoked when a Wallet Connect session request is received.
+  /// - onFailure: A callback invoked when an error occurs in the Wallet Connect connection.
+  /// - onDisconnect: A callback invoked when the Wallet Connect connection is disconnected.
+  /// - onEthSign: A callback invoked when an Ethereum transaction signing request is received.
+  /// - onEthSignTransaction: A callback invoked when an Ethereum transaction signing request is received.
+  /// - onEthSendTransaction: A callback invoked when an Ethereum transaction sending request is received.
+  /// - maticRpcUri: The RPC URI for the blockchain service provider to be used.
+  ///
+  /// Return:
+  /// This function returns a Future<WCAttributes> that contains a WCAttributes object
+  /// which includes the WCClient instance, WCSessionStore session storage, and Web3Client instance.
+  /// The Future value will be resolved once the initialization is complete.
+  ///
+  Future<WCAttributes> initWalletConnect({
+    SessionRequest? onSessionRequest,
+    SocketError? onFailure,
+    SocketClose? onDisconnect,
+    EthSign? onEthSign,
+    EthTransaction? onEthSignTransaction,
+    EthTransaction? onEthSendTransaction,
+    required String maticRpcUri,
```

Comment:
Change this variable to `rpcUri`. So it's not confusing whether it can accept only Polygon or any EVM Chain.